tests/phpunit/tests/webfonts-api/wpWebfontsController.php

```diff
@@ -0,0 +1,210 @@
+											->getMock();
+		$this->controller             = new WP_Webfonts_Controller(
+			$this->webfont_registry_mock,
+			$this->provider_registry_mock
+		);
+	}
+
+	/**
+	 * @covers WP_Webfonts_Controller::init
+	 *
+	 * @dataProvider data_init
+	 *
+	 * @param string $hook       Expected hook name.
+	 * @param bool   $did_action Whether the action fired or not.
+	 */
+	public function test_init( $hook, $did_action ) {
+		$this->provider_registry_mock
+			->expects( $this->once() )
+			->method( 'init' );
+
+		if ( $did_action ) {
+			do_action( 'wp_enqueue_scripts' );
+		}
+
+		$this->controller->init();
+
+		$this->assertSame(
+			10,
+			has_action( $hook, array( $this->controller, 'generate_and_enqueue_styles' ) )
+		);
+		$this->assertSame(
+			10,
+			has_action( 'admin_init', array( $this->controller, 'generate_and_enqueue_editor_styles' ) )
+		);
+	}
+
+	/**
+	 * Data provider.
+	 *
+	 * @return array
+	 */
+	public function data_init() {
+		return array(
+			'did_action fired'        => array(
+				'hook'       => 'wp_print_footer_scripts',
+				'did_action' => true,
+			),
+			'did_action did not fire' => array(
+				'hook'       => 'wp_enqueue_scripts',
+				'did_action' => false,
+			),
+		);
+	}
+
+	/**
+	 * By default, the Webfonts API should not request webfonts from
+	 * a remote provider. Test the permissions logic works as expected.
+	 *
+	 * @covers WP_Webfonts_Controller::generate_and_enqueue_styles
+	 *
+	 * @dataProvider data_generate_and_enqueue_editor_styles
+	 *
+	 * @param string $stylestyle_handle Handle for the registered stylesheet.
+	 */
+	public function test_generate_and_enqueue_styles_default( $stylesheet_handle ) {
+		/*
+		 * Set the stylesheet_handle property.
+		 * This is set in WP_Webfonts_Controller::init(); however, init is not part
+		 * of this test (as it has its own test).
+		 */
+		$property = new ReflectionProperty( $this->controller, 'stylesheet_handle' );
+		$property->setAccessible( true );
+		$property->setValue( $this->controller, $stylesheet_handle );
+
+		// Set up the provider mock.
+		$provider  = $this->getMockBuilder( 'WP_Webfonts_Google_Provider' )->getMock();
+		$providers = array(
+			'google' => $provider,
+		);
+		$this->provider_registry_mock
+			->expects( $this->once() )
+			->method( 'get_all_registered' )
+			->willReturn( $providers );
+		// The Google Fonts provider should never be called.
+		$provider
+			->expects( $this->never() )
+			->method( 'set_webfonts' );
+
+		// Fire the method being tested.
+		$this->controller->generate_and_enqueue_styles();
+		$this->expectOutputString( '' );
+		wp_print_styles( $stylesheet_handle );
+	}
+
+	/**
+	 * @covers WP_Webfonts_Controller::generate_and_enqueue_styles
+	 * @covers WP_Webfonts_Controller::generate_and_enqueue_editor_styles
+	 *
+	 * @dataProvider data_generate_and_enqueue_editor_styles
+	 *
+	 * @param string $stylestyle_handle Handle for the registered stylesheet.
```

Comment:
```suggestion
	 * @param string $stylesheet_handle Handle for the registered stylesheet.
```
This was missed when updating the parameter name in the function declaration.

Reply:
Thank you @costdev - fixed in https://github.com/WordPress/wordpress-develop/pull/1736/commits/92bd91195f7896404e666f8a091b7224903157e9.